accounting/transferwise/models.py

```diff
@@ -0,0 +1,181 @@
+
+    class Meta:
+        verbose_name = _('TransferWise Bulk Payment')
+        verbose_name_plural = _('TransferWise Bulk Payments')
+
+    def __str__(self):
+        """
+        Indicate who's sending this TransferWise bulk payment, and when.
+        """
+        return '{date}: {sender}'.format(date=self.date, sender=self.sender)
+
+    @property
+    def recipients(self):
+        """
+        Return a list of TransferWise recipients for the sender of this bulk payment.
+        """
+        # We get a list of recipients who are connected to the sender and have TransferWise recipient IDs.
+        hourly_rates = self.sender.client_hourly_rates.exclude(provider__bank_accounts__transferwise_recipient_id=None)
+        return [rate.provider for rate in hourly_rates]
+
+    @property
+    def csv_filename(self):
+        """
+        Return a CSV filename for this bulk payment.
+        """
+        return 'transferwise_bulk_payment_csv_{sender}_{date}.csv'.format(
+            sender=self.sender.user.username,
+            date=self.date.strftime("%Y-%m-%d"),
+        )
+
+    def create_payments(self):
+        """
+        Create the individual TransferWise payments needed for this bulk payment.
+
+        The chosen invoice is the one that matches this bulk payment's date's month for the given payment's
+        sender and recipient. If an invoice does not fit this criteria, it gets the latest unpaid one instead.
+
+        Note that the invoice is the source of truth for payments, and thus should exist before making a payment.
+
+        TODO: Besides creating an internal version of the payment for tracking certain data (i.e. recipient ID,
+        TODO: invoice, etc.), this should eventually actually call the API to create real transfers. At that stage,
+        TODO: we could then skip generating CSVs and create all the transfers in one go, and call them a 'bulk payment',
+        TODO: even if that terminology is only to represent how we're doing it internally.
+        """
+        for recipient in self.recipients:
+            try:
+                # Get an invoice matching this bulk payment's month & year.
+                invoice = recipient.provider_invoices.filter(
+                    client=self.sender,
+                    date__year=self.date.year,
+                    date__month=self.date.month,
+                    paid=False,
+                ).latest('date')
+            except Invoice.DoesNotExist:
+                # Fallback to the latest unpaid one, or just skip the recipient.
+                try:
+                    invoice = recipient.provider_invoices.filter(client=self.sender, paid=False).latest('date')
+                except Invoice.DoesNotExist:
+                    continue
+
+            try:
+                self.payments.create(bulk_payment=self, invoice=invoice)
+            except (IntegrityError, ValidationError):
+                continue
```

Comment:
Why do we ignore these exceptions?

Reply:
@antoviaque This is mostly in place because you can hit 'save' and create these same payments twice. So it just covers that possible mistake. This won't be needed when we put this into an automated job though.

Reply:
@UmanShahzad Could you add a TODO item there and an explanation of the scenario where this happens?

Reply:
@antoviaque Sure. :+1: